addons/project-source-button/viewer.html

```diff
@@ -0,0 +1,77 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8" />
+    <style>
+      #viewer {
+        width: 50vw;
+        height: 90vh;
+      }
+      #json-editor {
+        width: 95%;
+        height: 90%;
+      }
+      #json-code {
+        overflow: scroll;
+        width: 100%;
+        height: 90%;
+      }
+      html,
+      body {
+        margin: 0;
+        padding: 0;
+      }
+      #assets {
+        overflow: scroll;
+        width: 50vw;
+        height: 90vh;
+      }
+      #ctrl span {
+        font-size: 1.5rem;
+      }
+      #ctrl {
+        margin: 0 auto;
+        text-align: center;
+      }
+      #next,
+      #back {
+        background-color: grey;
+      }
+      #all-view {
+        background-color: #ddd;
+      }
+      #edit-mode {
+        background-color: azure;
+      }
+    </style>
+    <script src="../../libraries/common/cs/jszip.min.js"></script>
+    <link rel="stylesheet" href="../../libraries/common/cs/prism.css" />
+    <script src="../../libraries/common/cs/prism.js"></script>
+  </head>
+  <body>
+    <h1 id="h-title">Loding...</h1>
+    <div style="display: flex">
+      <div id="viewer">
+        <textarea id="json-editor" hidden></textarea>
+        <div id="json-code">
+          <pre class="line-numbers"><code class="language-json"></code></pre>
+        </div>
+        <div id="ctrl">
+          <button id="back">←</button>
+          <span>
+            <span id="page">1</span>
+            <span>/</span>
+            <span id="max-page">1</span>
+          </span>
+          <button id="next">→</button>
+          <br />
+          <button id="all-view">All view</button>
+          <button id="edit-mode">Edit/View</button>
```

Comment:
Could this be a checkbox or toggle instead?

Reply:
Solve it.